app/controllers/controllershelpers/HomeCardGenerator.scala

```diff
@@ -96,22 +96,25 @@ class HomeCardGenerator @Inject() (
       }
     }
 
-  def getSaAndItsaMergeCardOrSaWithoutUtrCard()(implicit
-    messages: Messages,
-    request: UserRequest[_]
+  def getSelfAssessmentCard()(implicit
+                              messages: Messages,
+                              request: UserRequest[_]
   ): Option[HtmlFormat.Appendable] = {
+
     val isItsaEnrolled = enrolmentsHelper.itsaEnrolmentStatus(request.enrolments).isDefined
-    val saView         = saAndItsaMergeView(
-      (current.currentYear + 1).toString,
-      isItsaEnrolled
-    )
+    val saView = saAndItsaMergeView((current.currentYear + 1).toString, isItsaEnrolled)
 
-    request.trustedHelper.isEmpty match {
-      case false                                  => None
-      case true if isItsaEnrolled || request.isSa => Some(saView)
-      case true if configDecorator.pegaEnabled    =>
-        Some(selfAssessmentRegistrationView()) // Temporary pegaEnabled condition
-      case true                                   => None // In the future, this will be removed when pegaEnabled condition is removed
+    if (request.trustedHelper.isDefined) {
```

Comment:
You could just map/ flatMap this. A None => None

Reply:
Are you suggesting to use fold for request.trustedHelper, or request.trustedHelper.flatMap(_ => None).orElse {
      if (isItsaEnrolled || request.isSa) {
        Some(saView)
      } else if (configDecorator.pegaEnabled) {
        // Temporary condition for Pega
        Some(selfAssessmentRegistrationView())
      } else {
        None
      }
    }

I wanted to keep all logic in one type like if elses
, hence did that. but can do what you are suugesting

Reply:
request.trustedHelper.flatMap{ _ =>
      if (isItsaEnrolled || request.isSa) {
        Some(saView)
      } else if (configDecorator.pegaEnabled) {
        // Temporary condition for Pega
        Some(selfAssessmentRegistrationView())
      } else {
        None
      }
    }

is one line less. However, I'm OK with the way you've done it too.

Reply:
isn't your code the opposite of what we would expect. It would go on to check for other things even if the trustedhelper is defined while we need it to return None

Reply:
Just stick with what you got Sandeep, it's fine